code/datums/mutations/holy_mutation/burdened.dm

```diff
@@ -0,0 +1,161 @@
+
+/datum/mutation/human/burdened/proc/organ_added_burden(mob/burdened, obj/item/organ/new_organ)
+	SIGNAL_HANDLER
+
+	if(istype(new_organ, /obj/item/organ/eyes))
+		var/obj/item/organ/eyes/new_eyes = new_organ
+		if(new_eyes.tint < TINT_BLIND) //unless you added unworking eyes (flashlight eyes), this is removing burden
+			burden_level -= 1
+			update_burden(FALSE)
+
+/datum/mutation/human/burdened/proc/organ_removed_burden(mob/burdened, obj/item/organ/old_organ)
+	SIGNAL_HANDLER
+
+	if(istype(old_organ, /obj/item/organ/eyes))
+		var/obj/item/organ/eyes/old_eyes = old_organ
+		if(old_eyes.tint < TINT_BLIND) //unless you were already blinded by them (flashlight eyes), this is adding burden!
+			burden_level += 1
+			update_burden(TRUE)
+
+/datum/mutation/human/burdened/proc/limbs_added_burden(obj/item/bodypart/limb_added, special, dismembered)
+	SIGNAL_HANDLER
+
+	if(special) //something we don't wanna consider, like instaswapping limbs
+		return
+	var/mob/living/carbon/burdened = limb_added.owner
+	if(burdened.bodyparts.len == 5) //adding a limb got you to chest, head, 3 limbs
+		burden_level -= 1 //which counts as removing burden
+		update_burden(FALSE)
+
+/datum/mutation/human/burdened/proc/limbs_removed_burden(obj/item/bodypart/limb_lost, special, dismembered)
+	SIGNAL_HANDLER
+
+	if(special) //something we don't wanna consider, like instaswapping limbs
+		return
+	var/mob/living/carbon/burdened = limb_lost.owner
+	if(burdened.bodyparts.len == 4) //adding a limb got you to chest, head, 2 limbs
+		burden_level += 1 //which counts as adding burden
+		update_burden(TRUE)
+
+/datum/mutation/human/burdened/proc/addict_added_burden(datum/addiction/new_addiction, datum/mind/addict_mind)
+	SIGNAL_HANDLER
+
+	if(addict_mind.active_addictions.len)
+		return //already did this
+	burden_level += 1 //you're addicted to something
+	update_burden(TRUE)
+
+/datum/mutation/human/burdened/proc/addict_removed_burden(datum/addiction/old_addiction, datum/mind/nonaddict_mind)
+	SIGNAL_HANDLER
+
+	if(!nonaddict_mind.active_addictions.len)
+		burden_level -= 1 //no longer addicted to anything
+		update_burden(FALSE)
+
+/datum/mutation/human/burdened/proc/mutation_added_burden(mob/living/carbon/burdened, datum/mind/addict_mind)
+	SIGNAL_HANDLER
+
+	var/bad_mutations = 0
+	for(var/datum/mutation/human/mutation as anything in burdened.dna.mutations)
+		if(mutation.quality == NEGATIVE)
+			bad_mutations++
+	if(bad_mutations == BAD_MUTATIONS_REQUIRED)
+		burden_level += 1 //you're badly mutated!
+		update_burden(TRUE)
+
+/datum/mutation/human/burdened/proc/mutation_removed_burden(mob/living/carbon/burdened, datum/mind/nonaddict_mind)
+	SIGNAL_HANDLER
+
+	var/bad_mutations = 0
+	for(var/datum/mutation/human/mutation as anything in burdened.dna.mutations)
+		if(mutation.quality == NEGATIVE)
+			bad_mutations++
+	if(bad_mutations == BAD_MUTATIONS_REQUIRED == 2) //looks bad but can only be checked if we are losing a mutation
```

Comment:
why are there two ==s in the same statement here?

Reply:
eek

Reply:
also, how does this interact with mutadone and such?